services/data-info/src/data_info/routes/permissions.clj

```diff
@@ -28,11 +28,34 @@
     :path-params [data-id :- DataIdPathParam]
     :tags ["data-by-id"]
 
-    (GET* "/permissions" [:as {uri :uri}]
-      :query [params StandardUserQueryParams]
-      :return DataItemPermissionsResponse
-      :summary "Lists Data Item Permissions"
-      :description (str
-  "Lists permissions for a data item."
-  (get-error-code-block "ERR_DOES_NOT_EXIST, ERR_NOT_READABLE, ERR_NOT_A_USER"))
-      (svc/trap uri perms/list-permissions params data-id))))
+    (context* "/permissions" []
+
+      (GET* "/" [:as {uri :uri}]
+        :query [params StandardUserQueryParams]
+        :return DataItemPermissionsResponse
+        :summary "List Data Item Permissions"
+        :description (str
+"Lists permissions for a data item."
+(get-error-code-block "ERR_DOES_NOT_EXIST, ERR_NOT_READABLE, ERR_NOT_A_USER"))
+        (svc/trap uri perms/list-permissions params data-id))
+
+      (PUT* "/:share-with/:permission" [:as {uri :uri}]
+        :path-params [share-with :- (describe NonBlankString "The user to grant permissions to.")
+                      permission :- (describe PermissionEnum "The permission level to grant.")]
+        :query [params StandardUserQueryParams]
+        :return DataItemPermissionsResponse
+        :summary "Grant Data Item Permissions"
+        :description (str
+"Grants access to a data item."
+(get-error-code-block "ERR_DOES_NOT_EXIST, ERR_NOT_READABLE, ERR_NOT_A_USER", "ERR_NOT_OWNER"))
```

Comment:
I suspect the last comma of this list was intended to be inside the string? (same below)